File: src/main/java/ast/ASTDotExpr.java
package ast;

public class ASTDotExpr extends ASTExpr {
    public ASTExpr left;
    public ASTExpr right;

    public ASTDotExpr(ASTFileLocation loc, ASTExpr left, ASTExpr right){
        super(loc);
        this.left = left;
        this.right = right;
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTDotExpr:");
        left.print(i+1);
        right.print(i+1);
    }
}

File: src/main/kotlin/ast/ASTDotExpr.kt
package ast

class ASTDotExpr(loc: ASTFileLocation, val left: ASTExpr, val right: ASTExpr) : ASTExpr(loc) {

    override fun isLValue(): Boolean {
        return true
    }

    override fun print(i: Int) {
        printIndent(i)
        println("ASTDotExpr:")
        left.print(i + 1)
        right.print(i + 1)
    }
}